src/magnifier.c

```diff
@@ -112,16 +112,26 @@ magnify(struct output *output, struct wlr_buffer *output_buffer, struct wlr_box
 		return;
 	}
 
+	/* Paste the magnified result back into the output buffer */
+	if (!tmp_texture) {
+		tmp_texture = wlr_texture_from_buffer(server->renderer, tmp_buffer);
+	}
+	if (!tmp_texture) {
+		wlr_log(WLR_ERROR, "Failed to allocate temporary texture");
+		return;
+	}
```

Comment:
This may cause us to run into the earlier `assert(tmp_texture)` when the dimension of the buffer change (tmp_buffer set but no tmp_texture). Suggest adding a `wlr_buffer_drop(tmp_buffer); tmp_buffer = NULL;` into the branch.